bentoml/yatai/db/stores/lock.py

```diff
@@ -0,0 +1,54 @@
+import enum
+
+from sqlalchemy import UniqueConstraint, Column, Integer, Enum
+from bentoml.exceptions import LockUnavailable
+from bentoml.yatai.db import Base
+
+
+class LOCK_STATUS(enum.Enum):
+    read_lock = 1
+    write_lock = 2
+
+
+class Lock(Base):
+    __tablename__ = 'locks'
+    __table_args__ = tuple(UniqueConstraint('resource_id', name='_resource_id_uc',))
+    id = Column(Integer, primary_key=True)
+    resource_id = Column(Integer, nullable=False)
+    lock_status = Column(Enum(LOCK_STATUS))
+
+
+class LockStore(object):
+    @staticmethod
+    def _find_lock(sess, resource_id):
+        lock_obj = sess.query(Lock).filter_by(resource_id=resource_id).first()
+        return lock_obj
+
+    @staticmethod
+    def acquire(sess, lock_type, resource_id):
+        lock = LockStore._find_lock(sess, resource_id)
+
+        # no lock found; free to acquire
+        if not lock:
+            lock = Lock()
+            lock.lock_status = lock_type
+            lock.resource_id = resource_id
+            sess.add(lock)
```

Comment:
https://docs.sqlalchemy.org/en/13/orm/session_api.html#sqlalchemy.orm.session.Session.add

> Its state will be persisted to the database on the next flush operation.

So the lock won't work between instances of Yatai?